src/app/keyboard/components/virtual-keyboard/virtual-keyboard.component.ts

```diff
@@ -53,34 +61,55 @@ export class VirtualKeyboardComponent implements OnChanges {
   private updateAvailableKeyBindings(): void {
     this.currentModifierKeystrokes = this.keystrokes.filter(keystroke => this.keyAliasService.isModifierKeyOrAliasedModifierKey(keystroke))
     this.keyBindingsFilteredByModifiers = this.keyBindings.filter(keyBinding => {
-      const modifierKeys: string[] = keyBinding.keys.filter(key => this.keyAliasService.isModifierKeyOrAliasedModifierKey(key))
-      if (modifierKeys.length !== this.currentModifierKeystrokes.length) {
-        return false
+      if (keyBinding.mappedKeys && this.doesModifiersMatchExclusively(this.getKeyBindingToDisplay(keyBinding))) {
+        return true
       }
-      const isModifiersPartialOfKeyBindingModifiers: boolean = this.currentModifierKeystrokes.every(keystroke =>
-        modifierKeys.some(modifierKey => this.keyAliasService.isKeyPartOfAlias(keystroke, modifierKey))
-      )
-      return isModifiersPartialOfKeyBindingModifiers
+      return this.doesModifiersMatchExclusively(keyBinding)
     })
   }
 
+  private doesModifiersMatchExclusively(keyBinding: StyledKeyBinding): boolean {
+    const modifierKeysForKeyBinding: string[] = keyBinding.keys.filter(key => this.keyAliasService.isModifierKeyOrAliasedModifierKey(key))
+
+    if (modifierKeysForKeyBinding.length !== this.currentModifierKeystrokes.length) {
+      return false
+    }
+
+    // TODO: Respect key order and exclusivity
```

Comment:
Is this planned to be part of this PR?

Reply:
I was not planning on doing it in this PR, since tit is not something that we use, but something that other parties might be interested in.